### DISCUSSION TITLE
Cannot create disambiguating relations with prisma client

### DISCUSSION THREAD

**Discussion by sagarPakhrin** [prisma/prisma#14937] [Q&A]
### Bug description

I have two models with disambiguating relations. 

```prisma
model Product {
  id          Int     @id @default(autoincrement())
  name        String?
  fees        Fee[]   @relation("fees")
  default_fee Fee?    @relation("default_fee")
}

model Fee {
  id             Int      @id @default(autoincrement())
  total          Float    @default(0)
  product_id     Int
  product        Product  @relation(name: "fees", fields: [product_id], references: [id], onDelete: Cascade)
  defaultable    Product? @relation(name: "default_fee", fields: [defaultable_id], references: [id])
  defaultable_id Int?     @unique
}
```

When I try to create a create a product with nested relations,
```ts
const product = await this.prisma.product.create({
  data: {
    name: 'product1',
    default_fee: {
      create: {
        total: 100,
      },
    },
  },
});
```

 it throws me error saying
```
Type '{ total: number; }' is not assignable to type '(Without<FeeCreateWithoutDefaultableInput, FeeUncheckedCreateWithoutDefaultableInput> & FeeUncheckedCreateWithoutDefaultableInput) | (Without<...> & FeeCreateWithoutDefaultableInput)'.
  Type '{ total: number; }' is not assignable to type 'Without<FeeUncheckedCreateWithoutDefaultableInput, FeeCreateWithoutDefaultableInput> & FeeCreateWithoutDefaultableInput'.
    Property 'product' is missing in type '{ total: number; }' but required in type 'FeeCreateWithoutDefaultableInput'.ts(2322)
index.d.ts(3389, 5): 'product' is declared here.
index.d.ts(3139, 5): The expected type comes from property 'create' which is declared here on type 'FeeUncheckedCreateNestedOneWithoutDefaultableInput | FeeCreateNestedOneWithoutDefaultableInput'
```
interestingly enough, breaking into multiple queries like this works.
```ts
const fee = await this.prisma.fee.create({
  data: {
    total: 100,
    defaultable_id: product.id,
    product_id: product.id
  },
});
```

Why is the first approach not working?

### How to reproduce

1. `git clone git@github.com:sagarPakhrin/prisma-disambiguating-relations.git`
2. `cd prisma-disambiguating-relations`
3. `npm i `
4. `npm run start:dev`

view `src/app.service.ts`

### Prisma information

I'm using prisma `v4` with nestjs

### Environment & setup

Environment & setup
    OS: ubuntu22
    Database: mysql
    Node.js version: 16

### Prisma Version

**Prisma Version**
`    "@prisma/client": "^4.2.1",`

**Comment by austincrim** [comment]
Hey @sagarPakhrin 👋,

The reason the first query does not work is that the `Fee.product` field is required in the `default_fee` property, but you are only providing the `total`.

This query worked for me:
```ts
await prisma.product.create({
  data: {
    name: 'product',
    default_fee: {
      create: {
        total: 100,
        product: {
          create: {}
        }
      }
    }
  }
})
```

**Reply by sagarPakhrin** [reply]
This query only seems to work because the name in my product is optional and since you passes `create` inside the `default_fee` it creates an extra product with all null values, which is not what I want.

When I have a mandatory field in product model it stops working

**Reply by lamasagar** [reply]
for now the solutions is to break the query down query into multiple step
```
const product = await this.prisma.product.create({
  data: {
    name: 'product1',
  },
});
await this.prisma.fee.create({
  data: {
    total: 100,
    product_id: product.id,
    defaultable_id: product.id,
  },
});
```

If its supposed to be this way by design, it would be great if it could be mentioned in the docs

**Reply by austincrim** [reply]
Ah, I see. That's a good piece of feedback, thank you for bringing it to our attention!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
